src/scalar.rs

```diff
@@ -59,6 +65,22 @@ impl PartialEq for Scalar {
     }
 }
 
+impl Ord for Scalar {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        let mut self_bytes = self.0;
+        let mut other_bytes = other.0;
+        &self_bytes.reverse();
+        &other_bytes.reverse();
+        self_bytes.cmp(&other_bytes)
```

Comment:
This will fail on some pairs, because scalars are internally stored in Montgomery form and will have a different ordering. I believe you intended to use `Scalar::to_bytes` instead?

Reply:
Done in b5ef8e2!